cmd/subnetcmd/deploy.go

```diff
@@ -187,62 +184,82 @@ func runDeploy(cmd *cobra.Command, args []string) error {
-	if err != nil {
-		return err
+func updateSubnetEVMGenesisChainID(genesisBytes []byte, newChainID uint) ([]byte, error) {
+	var genesisMap map[string]interface{}
+	if err := json.Unmarshal(genesisBytes, &genesisMap); err != nil {
+		return nil, err
 	}
-	ux.Logger.PrintToUser("Enter your subnet's ChainID. It can be any positive integer.")
-	var chainID *big.Int
-	if mainnetChainID != "" {
-		newChainID := new(big.Int)
-		newChainID, ok := newChainID.SetString(mainnetChainID, 10)
-		if !ok {
-			return errors.New("SetString: error")
-		}
-		chainID = newChainID
-	} else {
-		chainID, err = app.Prompt.CapturePositiveBigInt("ChainID")
-		if err != nil {
-			return err
-		}
+	configI, ok := genesisMap["config"]
+	if !ok {
+		return nil, fmt.Errorf("config field not found on genesis")
 	}
-	evmGenesis.Config.ChainID = chainID
-	jsonBytes, err := evmGenesis.MarshalJSON()
-	if err != nil {
-		return err
+	config, ok := configI.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("expected genesis config field to be a map[string]interface, found %T", configI)
 	}
-	var prettyJSON bytes.Buffer
-	err = json.Indent(&prettyJSON, jsonBytes, "", "    ")
+	config["chainId"] = float64(newChainID)
+	return json.MarshalIndent(genesisMap, "", "  ")
+}
+
+// updates sidecar with genesis mainnet id to use
+// given either by cmdline flag, original genesis id, or id obtained from the user
+func getSubnetEVMMainnetChainID(sc *models.Sidecar, subnetName string) error {
+	// get original chain id
+	evmGenesis, err := app.LoadEvmGenesis(subnetName)
 	if err != nil {
 		return err
 	}
-	return app.WriteGenesisMainnetFile(chain, prettyJSON.Bytes())
-}
-
-func handleMainnetChainID(chain string) error {
-	genesisMainnetPath := app.GetGenesisMainnetPath(chain)
-	_, err := os.ReadFile(genesisMainnetPath)
-	if err != nil && os.IsNotExist(err) {
+	if evmGenesis.Config == nil {
+		return fmt.Errorf("invalid subnet evm genesis format: config is nil")
+	}
+	if evmGenesis.Config.ChainID == nil {
+		return fmt.Errorf("invalid subnet evm genesis format: config chain id is nil")
+	}
+	originalChainID := evmGenesis.Config.ChainID.Uint64()
+	// handle cmdline flag if given
+	if mainnetChainID != 0 {
+		sc.SubnetEVMMainnetChainID = uint(mainnetChainID)
+	}
+	// prompt the user
+	if sc.SubnetEVMMainnetChainID == 0 {
 		useSameChainID := "Use same ChainID"
-		createNewGenesis := "Use new ChainID"
-		listOptions := []string{createNewGenesis, useSameChainID}
+		useNewChainID := "Use new ChainID"
+		listOptions := []string{useNewChainID, useSameChainID}
 		newChainIDPrompt := "Using the same ChainID for both Fuji and Mainnet could lead to a replay attack. Do you want to use a different ChainID?"
-		var decision string
-		if mainnetChainID == "" && os.Getenv(constants.SimulatePublicNetwork) == "" {
-			decision, err = app.Prompt.CaptureList(newChainIDPrompt, listOptions)
-			if err != nil {
-				return err
-			}
+		var (
+			err      error
+			decision string
```

Comment:
don't need this just use `decision, err := app.Prompt.CaptureList(newChainIDPrompt, listOptions)`